File: python/22_names_scores.py
def names_scores(names: list) -> int:
    sum: int = 0
    # iterate over list of names
    for i in range(0, len(names)):
        # iterate over characters in name
        for j in range(0, len(names[i])):
            # letter score: ord(letter) - ord('A') + 1
            # A -> 1, B -> 2, C -> 3, ..., Z -> 26
            # multiply letter score by alphabetical position of name in list, i + 1
            sum += (ord(names[i][j]) - ord('A') + 1) * (i + 1)
    return sum

if __name__ == "__main__":
    f = open("22_names.txt", 'r')
    # names are stored as a comma separated list on a single line
    names = f.read().replace("\"", "").strip().split(",")
    f.close()
    print(names_scores(sorted(names)))

File: cpp/22_names_scores.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include <cstring>
#include <algorithm>

int names_scores(std::vector<std::string> names){
    int total = 0;
    for (int i = 0; i < names.size(); i++){
        int score = 0;
        for (char c : names[i]){
            // A = 1, B = 2, C = 3, ..., Z = 26
            score += c - 'A' + 1;
        }
        // multiply score by alphabetical position of name in vector
        total += (i + 1) * score;
    }
    return total;
}

int main(){
    // read names from file and store in string
    std::ifstream file("/uploads/22_names.txt");
    std::string s;
    std::getline(file, s);
    file.close();
    
    // remove double quotes from string
    s.erase(std::remove(s.begin(), s.end(), '\"'), s.end());
    
    // tokenize string with comma delimiter
    std::vector<std::string> names;
    char *pch = std::strtok(&s[0], ",");
    while (pch != NULL){
        names.push_back(pch);
        pch = std::strtok(NULL, ",");
    }
    
    // sort vector of names alphabetically
    std::sort(names.begin(), names.end());
    std::cout << names_scores(names) << "\n";
}
